### ISSUE TITLE
[Repro] Sim10K → Cityscapes (RT-DETR R34) Stage2 AP50 peaks early then drops (configs attached)

### ISSUE THREAD

**Issue by sarahshi99** [Jeremy-lf/RT-DATR#6]
Thank you for releasing the code and sharing this work. I’m trying to reproduce the **Sim10K → Cityscapes** UDA setting with **RT-DETR + ResNet-34**, but my Stage2 training behavior does not match the expected trend.

I’m attaching my Stage1/Stage2 configs (`repro_stage1.yml`, `repro_stage2.yml`) and curve screenshots for quick checking. 

---

### Stage 1 (source-only on Sim10K)
- Config: `repro_stage1.yml` (based on `rtdetr_r34vd_6x_coco.yml`)
- GPUs: 2× A6000
- LR: 5e-5
- Epochs: 72
- Checkpoint: **best on Sim10K val**

### Stage 2 (UDA Sim10K → Cityscapes)
- Config: `repro_stage2.yml` (based on `da_rtdetr_r34_backbone_encoder_instance_dn_cmt_sim10k2city.yml`)
- GPUs: 2× A6000
- Batch size: 8
- LR: 1e-4 (adjusted together with my GPU/batch setting)
- Epochs: 72
- Initialization: Stage2 `pretrain_weights` points to **Stage1 best**

**What I changed** (only):
- GPU count
- batch size and the corresponding LR adjustment
- dataset paths
- save_dir  
Everything else is kept identical to the released configs.

---

## Observations (Cityscapes val)
- Stage2 **AP50 peaks very early** and then keeps decreasing:
  - **best AP50 = 59.9** (around the first ~15 epochs)
  - then gradually drops to **56.3** at the final epoch
- I also had an accidental run where Stage2 loaded **only backbone weights** (instead of full Stage1 checkpoint):
  - AP50 curve becomes **“rise then stable”** (no obvious late degradation),
  - best AP50 is lower (**58.7**), but training is much more stable.

Curves are attached.

<img width="2047" height="566" alt="Image" src="https://github.com/user-attachments/assets/29c0a67d-f63e-402e-838c-b14e74ac45d4" />

<img width="2047" height="294" alt="Image" src="https://github.com/user-attachments/assets/9d27188d-3991-4896-8f9f-1a735056ab0b" />

<img width="2048" height="250" alt="Image" src="https://github.com/user-attachments/assets/9583f278-b338-4ff9-92a3-7abc7ddb24c5" />

---

### Q1. Config alignment
Could you please check whether my `repro_stage1.yml` and `repro_stage2.yml` are aligned with your intended pipeline for Sim10K→City?
If not, what are the key config items you would suggest checking/modifying first?

### Q2. Stage1 → Stage2 initialization (`pretrain_weights`): full vs backbone/encoder
Could you confirm the recommended initialization behavior for Stage2?
- Should Stage2 load the **full Stage1 source-only checkpoint** as `pretrain_weights`?
- Or is it better to load **only backbone/encoder weights** from Stage1 for UDA stability?
- If "only backbone/encoder" is recommended, could you point out the correct config option / format to load partial weights?

(Reason: in my experiments, full Stage1 init gives higher early AP50 but degrades later; backbone-only init is more stable but slightly lower best AP50.)

### Q3. Official checkpoints for sanity check (if possible)
If convenient, could you share an official checkpoint for:
- Stage1 (Sim10K source-only), and/or
- Stage2 (Sim10K→City)
So I can verify whether my dataset/evaluation/environment is aligned (sanity check).

### Q4. Diagnosing “early peak then degradation”
In your experience, what are the most common causes when Stage2 reaches a good AP50 early but then keeps dropping?
Any suggestions on priority checks would be very helpful (pseudo-label collapse/noise, DA loss weights/gradients, BN statistics mismatch, LR schedule/warmup, etc.).

---

## Attachments
- `repro_stage1.yml`
- `repro_stage2.yml`

[repro_stage2.yml](https://github.com/user-attachments/files/24184825/repro_stage2.yml)
[repro_stage1.yml](https://github.com/user-attachments/files/24184826/repro_stage1.yml)

Thanks again for your time and guidance!

**Comment by Jeremy-lf** [comment]
replied via email
